lib/IRGen/GenStruct.cpp

```diff
@@ -334,12 +345,38 @@ namespace {
       }
     }
   };
+  
+  // A helper to pick the best field to use for extra inhabitants in a
+  // fixed-layout record.
+  template<typename FieldInfo>
+  static const FieldInfo *
+  selectFixedExtraInhabitantProvidingField(IRGenModule &IGM,
+                                           ArrayRef<FieldInfo> fields) {
+    unsigned mostExtraInhabitants = 0;
+    const FieldInfo *fieldWithMost = nullptr;
+
+    for (auto &field : fields) {
+      auto *ti = dyn_cast<FixedTypeInfo>(&field.getTypeInfo());
+      // If any field is non-fixed, we can't definitively pick a best one.
```

Comment:
The correct check would be isAlwaysFixedSize(), which takes all resilience domains into account; otherwise, this will produce different results across resilience boundaries. However, there's no way currently to replicate that check at runtime.

Could we just call the dynamic extra inhabitant value witness functions in this case?